tgui/packages/tgui/interfaces/MusicInstrument.tsx

```diff
@@ -0,0 +1,183 @@
+import { useBackend, useLocalState } from '../backend';
+import { Button, Flex, LabeledList, Box, Knob, Input } from '../components';
+import { Window } from '../layouts';
+
+type MusicInstrumentData = {
+  name: string;
+  notes: string[];
+};
+
+export const MusicInstrument = (props, context) => {
+  const { act, data } = useBackend<MusicInstrumentData>(context);
+  const { name, notes } = data;
+
+  const [noteKeysOrder, setNoteKeysOrder] = useLocalState(
+    context,
+    'keyboardBindingsOrder',
+    'zsxdcvgbhnjmq2w3er5t6y7ui9o0p+'.split('')
+  );
+
+  const [activeKeys, setActiveKeys] = useLocalState(context, 'keyboardActivekeys', new Array(notes.length));
```

Comment:
`activeKeys` being an array when you're mostly using it as a map of key index to whether they're pressed or not seems a little odd. I think it's probably fine as-is, but sanity check on whether using a map-type structure (`{ 0: false, 1: true, ... }`) might make more sense for easy checking/updating.

Reply:
Alright, I'll change it. Thought of the piano as an array so that's why I thought it would make sense here too :)